src/Symfony/Component/VarDumper/Caster/CurlCaster.php

```diff
@@ -0,0 +1,30 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\VarDumper\Caster;
+
+use Symfony\Component\VarDumper\Cloner\Stub;
+
+/**
+ * @author Alexandre Daubois <alex.daubois@gmail.com>
+ */
+class CurlCaster
+{
+    public static function castCurl(\CurlHandle $h, array $a, Stub $stub, bool $isNested): array
+    {
+        return curl_getinfo($h);
+    }
+
+    public static function castCurlMulti(\CurlMultiHandle $h, array $a, Stub $stub, bool $isNested): array
```

Comment:
this one should be removed, curl_multi_info_read is not appropriate for casters

Reply:
I'm not sure to understand why? It is handy to have the different handles contained in the multihandle, what does make it not appropriate?

Reply:
See the description of the function: https://www.php.net/manual/en/function.curl-multi-info-read.php

This is an impure function changing the state of the queued message. Casters must not have side effects.